File: flutter/complex_app/lib/services/mock_comment_service.dart
import '../models/comment.dart';
import '../models/user.dart';

class MockCommentService {
  static final List<User> _users = List.generate(
    100,
    (i) => User(
      id: '${i + 1}',
      name: 'User${i + 1}',
      email: 'user${i + 1}@example.com',
    ),
  );

  static final List<Comment> _comments = List.generate(
    10000,
    (i) => Comment(
      id: '${i + 1}',
      bookId: '${(i % 1000) + 1}',
      author: _users[i % _users.length].name,
      content: 'This is a comment #${i + 1} for book ${(i % 1000) + 1}.',
      createdAt: DateTime.now().subtract(Duration(minutes: i)),
    ),
  );

  static List<Comment> getCommentsForBook(String bookId) =>
      _comments.where((c) => c.bookId == bookId).toList();

  static void addComment(Comment comment) {
    _comments.insert(0, comment);
  }
}

File: kotlin_multiplatform/complex_app/composeApp/src/commonMain/kotlin/com/example/complex_app/services/MockCommentService.kt
package com.example.complex_app.services

import com.example.complex_app.models.Comment
import kotlin.random.Random

object MockCommentService {
    private val comments = List(10000) {
        Comment(
            id = it.toString(),
            bookId = (it % 1000).toString(),
            userId = (it % 100).toString(),
            content = "Comment $it on Book ${(it % 1000)}",
            timestamp = System.currentTimeMillis() - Random.nextLong(0, 10000000)
        )
    }
    fun getCommentsForBook(bookId: String): List<Comment> = comments.filter { it.bookId == bookId }
    fun addComment(comment: Comment) { /* no-op for mock */ }
}
